crates/bevy_asset/src/asset_server.rs

```diff
@@ -781,8 +781,11 @@ mod test {
             asset_server.get_handle_untyped(id)
         }
 
-        fn get_asset(id: impl Into<HandleId>, world: &World) -> Option<&PngAsset> {
-            world.resource::<Assets<PngAsset>>().get(id.into())
+        fn get_asset<'world>(
```

Comment:
Do we need the explicit lifetime here?

Reply:
Yes, since both `world` and `id` are references, lifetime inference can't detect which one is correct. 
It works in the source methods because inference has special heuristics for `self` references.